packages/nimbus/src/components/date-input/date-input.recipe.ts

```diff
@@ -0,0 +1,128 @@
+import { defineSlotRecipe } from "@chakra-ui/react";
+
+/**
+ * Recipe configuration for the DateInput component.
+ * Defines the styling variants and base styles using Chakra UI's recipe system.
+ */
+export const dateInputSlotRecipe = defineSlotRecipe({
+  // Unique class name prefix for the component
+  className: "nimbus-date-input",
+  slots: ["root", "segmentGroup", "segment"],
+
+  // Base styles applied to all instances of the component
+  base: {
+    root: {
+      display: "inline-block",
+    },
+    segmentGroup: {
+      display: "flex",
+      borderRadius: "200",
+      bg: "transparent",
+      boxShadow: "inset 0 0 0 var(--border-width) var(--border-color)",
+      width: "full",
+      alignItems: "center",
+
+      "&[data-focus-within='true']": {
+        // TODO: can't use focusRing prop, find other solution (helper, util, etc.)
+        outlineWidth: "var(--focus-ring-width)",
+        outlineColor: "var(--focus-ring-color)",
+        outlineStyle: "var(--focus-ring-style)",
+        outlineOffset: "var(--focus-ring-offset)",
+      },
+      "&[data-invalid='true']": {
+        "--border-color": "colors.critical.7",
+        color: "critical.11",
```

Comment:
Should we also change the border width for the input here when it is invalid to `sizes.50`. Considering @FilPob [feedback](https://www.figma.com/design/AvtPX6g7OGGCRvNlatGOIY?node-id=3889-9483&m=dev#1289426361) on inputs? 
```suggestion
        color: "critical.11",
        "--border-width": "sizes.50"
```